### ISSUE TITLE
Use String Formatting (Placeholders) in Localization

### ISSUE THREAD

**Issue by VaslD** [HDInnovations/UNIT3D-Community-Edition#840]
### Issue Type
- [x] Enhancement

### End User Info
-- UNIT3D Version:

-- Laravel Version:

-- PHP Version:

-- Database Driver & Version:

-- Web Server Driver & Version:

-- OS Driver and Version:

### Expected Behavior:

Currently the project seems to be using string replacement for localization. For example, the English text "Are you *USERNAME*?" will have "Are you" extracted for localization, and the final translation in another language would be "~~some unrecognizable non-Latin characters~~ *USERNAME*?"

*This is very, very wrong!*

Many languages don't follow English language grammar (precisely the sentence structure popular in this language family). In some language, "Are you *USERNAME*?" can be and must only be translated to, say, "~~some unrecognizable non-Latin characters~~ *USERNAME* ~~some more unrecognizable non-Latin characters~~?" Or worse, "*USERNAME*, ~~some unrecognizable non-Latin characters~~?"

By using string replacement, although the translation for "Are you" is 100% correct, the entire sentence may not be understandable by native speakers.

In practical i18n, developers use string formatting to extract localizable resources. "Are you *USERNAME*?" should be extracted as a whole, to something like "Are you {0}?", with `{0}` serves as a placeholder to plug the username back in at runtime. (`{0}` or `%1s`, etc. depends on the programming language.) This way translators can deal with sentence structure discreetly, and rewrite the entire sentence as they see fit.

This conversion may require a great deal of additional work on the source code. But it's worth it. (Source: trust me 🤣)

### Steps To Reproduce The Problem:
1.
2.
3.

**Comment by HDVinnie** [comment]
Can you provide some images? I don't really get your point as parameters are passed to the multilingual string. We use laravel default localization.

https://laravel.com/docs/5.8/localization

Feel free to make a PR

**Comment by VaslD** [comment]
### So these strings are what I was talking about (bad examples):
<img width="960" alt="Annotation 2019-08-02 095140" src="https://user-images.githubusercontent.com/3415065/62338601-ac360580-b50b-11e9-876d-da9650781395.png">
<img width="960" alt="Annotation 2019-08-02 095251" src="https://user-images.githubusercontent.com/3415065/62338607-af30f600-b50b-11e9-9fd1-ff93ea5d4713.png">

I suspected that the name of the user/site are concatenated at the end/beginning of the string after localization. If I was right, doing it this way makes it hard to localize these strings properly in some languages. Maybe not for these particular sentences. It's just NOT a good way to i18n.

### Something like this is a good way to go:
<img width="580" alt="Annotation 2019-08-02 095431" src="https://user-images.githubusercontent.com/3415065/62338838-7e04f580-b50c-11e9-896b-36f9694f8bce.png">

Basically the author writes out sentences with full context and use placeholders to plug in variables at runtime, not assuming the name/value/etc. will always be concatenated at the beginning or the end of a sentence.

Back to the bad examples, to correct them, one would change to-localize string to:

- **Are you :user?**
- **:site is a closed community...**
- **This torrent is featured until :datetime**

and so on...

I'll see what I can do about PR but I don't fully understand the language and technology this project uses. I think I should stick to helping with localization for now.

**Comment by HDVinnie** [comment]
I see what you mean now. To be honest multilingual support is at the very bottom of my list.

**Comment by diamant-x** [comment]
This looks like already available in Laravel multilingual framework so is a matter of rewritting the sentences: https://laravel.com/docs/5.8/localization#replacing-parameters-in-translation-strings

**Comment by HDVinnie** [comment]
Pull Requests are welcome.

**Comment by diamant-x** [comment]
I did in the past but due to the high number of changes and This https://github.com/HDInnovations/UNIT3D-Community-Edition/issues/803 issue I don't see it that worth it as localisations might get outdated or changed in a few days. 
But I welcome anyone up for the task too.
